salt/modules/debian_ip.py

```diff
@@ -1498,6 +1498,13 @@ def _write_file_routes(iface, data, folder, pattern):
     '''
     Writes a file to disk
     '''
+    # Since the entire directory is executed with matching pattern
+    # ^[a-zA-Z0-9_-]+$ not all filenames are legitimate.
+    # In order to make routes work for vlan interfaces (e.g. bond0.12)
+    # the dots in the iface name will be replaced by underscores, so it
+    # will be properly executed
```

Comment:
Well, this patch is specifically for routes, not for interface files that are referenced or included via /etc/network/interfaces.

I have not tested the source-directory thing for interface files, but it could be that it needs to be there as well. In our environment all interfaces are configured in /etc/network/interfaces